app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,38 @@
+class AccountActivationsController < ApplicationController
+  before_action :load_user_for_activation, only: [:edit]
+  before_action :check_user_can_be_activated, only: [:edit]
+
+  def edit
+    activate_and_log_in_user
+  end
+
+  private
+
+  def load_user_for_activation
+    @user = User.find_by(email: params[:email])
+    return if @user
+
+    flash[:danger] = t(".invalid_activation_link")
+    redirect_to root_url
+  end
+
+  def check_user_can_be_activated
+    unless user_can_be_activated?
+      flash[:danger] = t(".invalid_activation_link")
+      redirect_to root_url
+    end
```

Comment:
```suggestion
    return if user_can_be_activated?
    
    flash[:danger] = t(".invalid_activation_link")
    redirect_to root_url
```
Viết return if cho clean code e nha

Reply:
done